### PR TITLE
Apply template icons

### LINKED ISSUE #628: Element template icon can be applied + updated via template to an element
__What should we do?__

Implement element templates icon support in the `cloud-element-templates` provider

* [ ] the update command can handle icons
  * [ ] property is applied from template
  * [ ] property is removed on update (no template)
  * [ ] property is updated on update (new template)
* [ ] the element factory can handle icons
* [ ] consider: validate - icon can be only defined for tasks

__Why should we do it?__

Child of https://github.com/bpmn-io/internal-docs/issues/445

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #628
Depends on https://github.com/camunda/element-templates-json-schema/pull/51
Depends on https://github.com/camunda-cloud/zeebe-bpmn-moddle/pull/24
Depends on https://github.com/bpmn-io/element-templates-icons-renderer/pull/2

This does not include displaying the icon in the properties panel, that's a follow up: https://github.com/bpmn-io/bpmn-js-properties-panel/pull/650

----

Script to try it out

```sh
npx @bpmn-io/sr bpmn-io/bpmn-js-properties-panel#628-apply-template-icons -c "npm run start:cloud-templates"
```

image

### CODE REVIEW

File: src/provider/cloud-element-templates/cmd/ChangeElementTemplateHandler.js [hunk 1]
```diff
@@ -99,6 +103,52 @@ export default class ChangeElementTemplateHandler {
     });
   }
 
+  _updateZeebeModelerTemplateIcon(element, newTemplate) {
+    const bpmnFactory = this._bpmnFactory,
+          commandStack = this._commandStack;
+
+    const icon = newTemplate && newTemplate.icon;
+
+    // remove existing icon when none new set
```
[the remark this answers is not in the document]
Comment (nikku) [reply src/provider/cloud-element-templates/cmd/ChangeElementTemplateHandler.js:112]: Exactly. Icon shall be removed once the unlink happens.

[the remark this answers is not in the document]
Comment (nikku) [reply src/provider/cloud-element-templates/cmd/ChangeElementTemplateHandler.js:112]: Everything else is miss-use, or a bug in our code. That is how I see it.

**Comment by pinussilvestrus** [comment]
Ready for review now.

**Review by barmac — APPROVED** [review]
LGTM

**Pull request merged**